samples/cpp/dft.cpp

```diff
@@ -1,86 +0,0 @@
-#include "opencv2/core.hpp"
```

Comment:
`cv::dft()` has code snipped, but it's embedded into header. It means that it's not built by CI. I propose to move the file to snippets, but change it's content to `convolveDFT` from core.hpp header file.

Reply:
@asmorkalov So should we change the contents of this sample to the contents of https://github.com/opencv/opencv/blob/96a8e6d76c6351ba61a13d9f01ab7c9cd79f983f/modules/core/include/opencv2/core.hpp#L2216 

and move this sample to snippets directory and change its reference to 
`@example samples/cpp/snippets/dft.cpp` in core.hpp?